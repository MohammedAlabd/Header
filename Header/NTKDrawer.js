import React from "react";
import { makeStyles } from "@material-ui/core/styles";
import ExpandLess from "@material-ui/icons/ExpandLess";
import { Drawer, List, IconButton } from "@material-ui/core";
import NestedListItem from "./NestedListItem";
import SingleListItem from "./SingleListItem";

const drawerWidth = "auto";

const useStyles = makeStyles((theme) => ({
  drawer: {
    width: drawerWidth,
    flexShrink: 0,
  },
  drawerPaper: {
    width: drawerWidth,
  },
  drawerHeader: {
    display: "flex",
    alignItems: "center",
    padding: theme.spacing(0, 1),
    // necessary for content to be below app bar
    ...theme.mixins.toolbar,
    justifyContent: "flex-end",
  },
}));

export default function NTKDrawer({
  pages,
  isDrawerOpen,
  handleDrawerClose,
  currentPagePath,
}) {
  const classes = useStyles();

  return (
    <Drawer
      className={classes.drawer}
      variant="persistent"
      anchor="top"
      open={isDrawerOpen}
      classes={{ paper: classes.drawerPaper }}
    >
      <div className={classes.drawerHeader}>
        <IconButton onClick={handleDrawerClose}>
          <ExpandLess />
        </IconButton>
      </div>
      <List>
        {pages.map((page) => {
          if (page.list) {
            return (
              <NestedListItem
                key={page.url}
                handleDrawerClose={handleDrawerClose}
                currentPagePath={currentPagePath}
                page={page}
              />
            );
          } else {
            return (
              <SingleListItem
                key={page.url}
                handleDrawerClose={handleDrawerClose}
                currentPagePath={currentPagePath}
                page={page}
              />
            );
          }
        })}
      </List>
    </Drawer>
  );
}
