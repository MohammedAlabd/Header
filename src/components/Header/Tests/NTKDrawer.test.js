import React from "react";
import NTKDrawer from "../NTKDrawer";
import "@testing-library/jest-dom";
import { render } from "@testing-library/react";
import { MemoryRouter } from "react-router-dom";

it("matches snapshot", () => {
  const pages = [
    { name: "Page1", url: "page1-url", list: false },
    {
      name: "Page2",
      url: "Page2-url",
      list: [{ name: "SubPage2", url: "subPage2-url" }],
    },
  ];
  const tree = render(
    <MemoryRouter>
      <NTKDrawer
        pages={pages}
        currentPagePath={"/page1-url"} //to make the color of Page1 button primary
        isDrawerOpen={true}
        handleDrawerClose={jest.fn}
      />
    </MemoryRouter>
  );

  expect(tree).toMatchSnapshot();
});
