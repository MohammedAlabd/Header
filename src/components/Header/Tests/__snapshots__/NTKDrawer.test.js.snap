// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="MuiDrawer-root MuiDrawer-docked makeStyles-drawer-1"
      >
        <div
          class="MuiPaper-root MuiDrawer-paper makeStyles-drawerPaper-2 MuiDrawer-paperAnchorTop MuiDrawer-paperAnchorDockedTop MuiPaper-elevation0"
        >
          <div
            class="makeStyles-drawerHeader-3"
          >
            <button
              class="MuiButtonBase-root MuiIconButton-root"
              tabindex="0"
              type="button"
            >
              <span
                class="MuiIconButton-label"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M12 8l-6 6 1.41 1.41L12 10.83l4.59 4.58L18 14z"
                  />
                </svg>
              </span>
              <span
                class="MuiTouchRipple-root"
              />
            </button>
          </div>
          <ul
            class="MuiList-root MuiList-padding"
          >
            <a
              class="makeStyles-textDecoration-69"
              href="/page1-url"
            >
              <div
                aria-disabled="false"
                class="MuiButtonBase-root MuiListItem-root MuiListItem-gutters MuiListItem-button"
                role="button"
                tabindex="0"
              >
                <h6
                  class="MuiTypography-root MuiTypography-subtitle1 MuiTypography-colorPrimary"
                >
                  PAGE1
                </h6>
                <span
                  class="MuiTouchRipple-root"
                />
              </div>
            </a>
            <div
              aria-disabled="false"
              class="MuiButtonBase-root MuiListItem-root MuiListItem-gutters MuiListItem-button"
              data-testid="openMenuButton"
              role="button"
              tabindex="0"
            >
              <h6
                class="MuiTypography-root MuiTypography-subtitle1 MuiTypography-colorSecondary"
              >
                PAGE2
              </h6>
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-colorSecondary"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M16.59 8.59L12 13.17 7.41 8.59 6 10l6 6 6-6z"
                />
              </svg>
              <span
                class="MuiTouchRipple-root"
              />
            </div>
          </ul>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="MuiDrawer-root MuiDrawer-docked makeStyles-drawer-1"
    >
      <div
        class="MuiPaper-root MuiDrawer-paper makeStyles-drawerPaper-2 MuiDrawer-paperAnchorTop MuiDrawer-paperAnchorDockedTop MuiPaper-elevation0"
      >
        <div
          class="makeStyles-drawerHeader-3"
        >
          <button
            class="MuiButtonBase-root MuiIconButton-root"
            tabindex="0"
            type="button"
          >
            <span
              class="MuiIconButton-label"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M12 8l-6 6 1.41 1.41L12 10.83l4.59 4.58L18 14z"
                />
              </svg>
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </button>
        </div>
        <ul
          class="MuiList-root MuiList-padding"
        >
          <a
            class="makeStyles-textDecoration-69"
            href="/page1-url"
          >
            <div
              aria-disabled="false"
              class="MuiButtonBase-root MuiListItem-root MuiListItem-gutters MuiListItem-button"
              role="button"
              tabindex="0"
            >
              <h6
                class="MuiTypography-root MuiTypography-subtitle1 MuiTypography-colorPrimary"
              >
                PAGE1
              </h6>
              <span
                class="MuiTouchRipple-root"
              />
            </div>
          </a>
          <div
            aria-disabled="false"
            class="MuiButtonBase-root MuiListItem-root MuiListItem-gutters MuiListItem-button"
            data-testid="openMenuButton"
            role="button"
            tabindex="0"
          >
            <h6
              class="MuiTypography-root MuiTypography-subtitle1 MuiTypography-colorSecondary"
            >
              PAGE2
            </h6>
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-colorSecondary"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M16.59 8.59L12 13.17 7.41 8.59 6 10l6 6 6-6z"
              />
            </svg>
            <span
              class="MuiTouchRipple-root"
            />
          </div>
        </ul>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
